modules/motions/LineTrace.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * @file   LineTrace.cpp
+ * @brief  ライントレースの親クラス
+ * @author miyahara046
+ */
+
+#include "LineTrace.h"
+using namespace std;
+
+LineTrace::LineTrace(Robot& _robot, double _targetSpeed, int _targetBrightness,
+                     const PidGain& _pidGain, bool& _isLeftEdge)
+  : Motion(_robot),
+    targetSpeed(_targetSpeed),
+    targetBrightness(_targetBrightness),
+    pidGain(_pidGain),
+    isLeftEdge(_isLeftEdge)
+
+{
+}
+
+void LineTrace::run()
+{
+  Pid pid(pidGain.kp, pidGain.ki, pidGain.kd, targetBrightness);
+
+  // 初期値を代入
+
+  initDistance = Mileage::calculateMileage(robot.getMotorControllerInstance().getRightMotorCount(),
+                                           robot.getMotorControllerInstance().getLeftMotorCount());
+
+  // 事前条件を判定する
+  if(!isMetPreCondition(targetSpeed)) {
+    return;
+  }
+
+  // 左右で符号を変える
+  int edgeSign = isLeftEdge ? -1 : 1;
+
+  // 呼び出し時の走行距離
+  initLeftMileage
+      = Mileage::calculateWheelMileage(robot.getMotorControllerInstance().getLeftMotorCount());
+  initRightMileage
+      = Mileage::calculateWheelMileage(robot.getMotorControllerInstance().getRightMotorCount());
+
+  int logIntervalCount = 0;  // 走行ログを取得するタイミングを計るための変数
+
+  // 継続条件を満たしている間ループ
+  while(isMetContinuationCondition()) {
+    // 初期pwm値を計算
+    double baseRightPwm = robot.getMotorControllerInstance().getRightMotorPower();
+    double baseLeftPwm = robot.getMotorControllerInstance().getLeftMotorPower();
+
+    // PIDで旋回値を計算
+    double turningPwm = pid.calculatePid(robot.getColorSensorInstance().getReflection()) * edgeSign;
+
+    // モータのPWM値をセット（前進の時0を下回らないように，後進の時0を上回らないようにセット）
+    double rightPwm = baseRightPwm > 0.0 ? max(baseRightPwm - turningPwm, 0.0)
+                                         : min(baseRightPwm + turningPwm, 0.0);
+    double leftPwm = baseLeftPwm > 0.0 ? max(baseLeftPwm + turningPwm, 0.0)
+                                       : min(baseLeftPwm - turningPwm, 0.0);
+    robot.getMotorControllerInstance().setRightMotorPower(rightPwm);
+    robot.getMotorControllerInstance().setLeftMotorPower(leftPwm);
+
+    // 10ms待機
```

Comment:
今回は，公式APIがマイクロ秒になってる．入力も三桁増しかな．
```suggestion
    // 10ms待機
```